### PR TITLE
Case 22224: Midi JSDoc

### PR DESCRIPTION
**Pull request by ctrlaltdavid**
Namepaces:
- Midi

Primary Manuscript case: https://highfidelity.manuscript.com/f/cases/22224/JSDoc-Midi

Other Manuscript cases addressed: None

Note: No executable code changes.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18819/HighFidelity-Beta-PR15422-6837d7e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18819/HighFidelity-Beta-Interface-PR15422-6837d7e.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18819/HighFidelity-Beta-PR15422-6837d7e.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18819/HighFidelity-Beta-Interface-PR15422-6837d7e.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18819/HighFidelity-Beta-PR15422-6837d7e.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18819/HighFidelity-Quest-Beta-PR15422-6837d7e.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18970/HighFidelity-Beta-PR15422-62cf427.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18970/HighFidelity-Beta-Interface-PR15422-62cf427.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18970/HighFidelity-Beta-PR15422-62cf427.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18970/HighFidelity-Quest-Beta-PR15422-62cf427.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18970/HighFidelity-Beta-PR15422-62cf427.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18970/HighFidelity-Beta-Interface-PR15422-62cf427.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18973/HighFidelity-Beta-PR15422-e7eaa78.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18973/HighFidelity-Beta-Interface-PR15422-e7eaa78.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18973/HighFidelity-Beta-PR15422-e7eaa78.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18973/HighFidelity-Quest-Beta-PR15422-e7eaa78.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18973/HighFidelity-Beta-PR15422-e7eaa78.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18973/HighFidelity-Beta-Interface-PR15422-e7eaa78.exe

File: libraries/midi/src/Midi.h [hunk 1]
```diff
@@ -49,139 +55,171 @@ class Midi : public QObject, public Dependency {
...
     /**jsdoc
+     * Enables or disables note off events.
      * @function Midi.typeNoteOffEnable
-     * @param {boolean} enable
+     * @param {boolean} enable - <code>true</code> to enable, <code>false</code> to disable.
      */
     Q_INVOKABLE void typeNoteOffEnable(bool enable);
 
     /**jsdoc
+     * Enables or disabled note on events.
```
Comment (nimisha20) [inline libraries/midi/src/Midi.h]:
```suggestion
     * Enables or disables note on events.
```

File: libraries/midi/src/Midi.h [hunk 2]
```diff
@@ -49,139 +55,171 @@ class Midi : public QObject, public Dependency {
...
      * @function Midi.typeNoteOnEnable
-     * @param {boolean} enable
+     * @param {boolean} enable - <code>true</code> to enable, <code>false</code> to disable.
      */
     Q_INVOKABLE void typeNoteOnEnable(bool enable);
 
     /**jsdoc
+     * Enables or disabled ply key pressure events.
```
Comment (nimisha20) [inline libraries/midi/src/Midi.h]:
```suggestion
     * Enables or disables poly key pressure events.
```

File: libraries/midi/src/Midi.h [hunk 3]
```diff
@@ -49,139 +55,171 @@ class Midi : public QObject, public Dependency {
...
      * @function Midi.typePolyKeyPressureEnable
-     * @param {boolean} enable
+     * @param {boolean} enable - <code>true</code> to enable, <code>false</code> to disable.
      */
     Q_INVOKABLE void typePolyKeyPressureEnable(bool enable);
 
     /**jsdoc
+     * Enables or disabled control change events.
```
Comment (nimisha20) [inline libraries/midi/src/Midi.h]:
```suggestion
     * Enables or disables control change events.
```

File: libraries/midi/src/Midi.h [hunk 4]
```diff
@@ -49,139 +55,171 @@ class Midi : public QObject, public Dependency {
...
      * @function Midi.typeControlChangeEnable
-     * @param {boolean} enable
+     * @param {boolean} enable - <code>true</code> to enable, <code>false</code> to disable.
      */
     Q_INVOKABLE void typeControlChangeEnable(bool enable);
 
     /**jsdoc
+     * Enables or disabled program change events.
```
Comment (nimisha20) [inline libraries/midi/src/Midi.h]:
```suggestion
     * Enables or disables program change events.
```

File: libraries/midi/src/Midi.h [hunk 5]
```diff
@@ -49,139 +55,171 @@ class Midi : public QObject, public Dependency {
...
     /**jsdoc
+     * Enables or disables channel pressure events.
      * @function Midi.typeChanPressureEnable
-     * @param {boolean} enable
+     * @param {boolean} enable - <code>true</code> to enable, <code>false</code> to disable.
      */
     Q_INVOKABLE void typeChanPressureEnable(bool enable);
 
     /**jsdoc
+     * Enables or disabled pitch bend events.
```
Comment (nimisha20) [inline libraries/midi/src/Midi.h]:
```suggestion
     * Enables or disables pitch bend events.
```

File: libraries/midi/src/Midi.cpp [hunk 6]
```diff
@@ -261,6 +261,61 @@ void Midi::MidiCleanup() {
 }
 #endif
 
+/**jsdoc
+ * A MIDI message.
+ * <p><strong>Warning:</strong> The <code>status</code> property is NOT a MIDI status value.</p>
+ * @typedef {object} Midi.MidiMessage
+ * @property {number} device - Device number.
+ * @property {Midi.RawMidiMessage} raw - Raw MIDI messag.
```
Comment (nimisha20) [inline libraries/midi/src/Midi.cpp]:
```suggestion
 * @property {Midi.RawMidiMessage} raw - Raw MIDI message.
```

File: libraries/midi/src/Midi.h [hunk 7]
```diff
@@ -49,139 +55,171 @@ class Midi : public QObject, public Dependency {
...
     Q_INVOKABLE void blockMidiDevice(QString name, bool output);
 
     /**jsdoc
-     * Unblock an input/output by name.
+     * Unblocks a MIDI device's input or output.
      * @function Midi.unblockMidiDevice
-     * @param {string} name
-     * @param {boolean} output
+     * @param {string} name- The name of the MIDI device to unblock.
+     * @param {boolean} output -  <code>true</code> to block the device's output, <code>false</code> to block its input.
```
Comment (nimisha20) [inline libraries/midi/src/Midi.h]:
```suggestion
     * @param {boolean} output -  <code>true</code> to unblock the device's output, <code>false</code> to unblock its input.
```

File: libraries/midi/src/Midi.h (hunk 1 above)
Comment (ctrlaltdavid) [reply libraries/midi/src/Midi.h]: Done.

File: libraries/midi/src/Midi.h (hunk 2 above)
Comment (ctrlaltdavid) [reply libraries/midi/src/Midi.h]: Done.

File: libraries/midi/src/Midi.h (hunk 3 above)
Comment (ctrlaltdavid) [reply libraries/midi/src/Midi.h]: Done.

File: libraries/midi/src/Midi.h (hunk 4 above)
Comment (ctrlaltdavid) [reply libraries/midi/src/Midi.h]: Done.

File: libraries/midi/src/Midi.h (hunk 5 above)
Comment (ctrlaltdavid) [reply libraries/midi/src/Midi.h]: Done.

File: libraries/midi/src/Midi.cpp (hunk 6 above)
Comment (ctrlaltdavid) [reply libraries/midi/src/Midi.cpp]: Done.

File: libraries/midi/src/Midi.h (hunk 7 above)
Comment (ctrlaltdavid) [reply libraries/midi/src/Midi.h]: Done.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19186/HighFidelity-Beta-PR15422-95e45a5.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19186/HighFidelity-Beta-Interface-PR15422-95e45a5.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19186/HighFidelity-Beta-PR15422-95e45a5.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19186/HighFidelity-Quest-Beta-PR15422-95e45a5.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19186/HighFidelity-Beta-PR15422-95e45a5.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19186/HighFidelity-Beta-Interface-PR15422-95e45a5.exe

**Pull request merged**